render.c

```diff
@@ -93,40 +117,71 @@ cairo_surface_t *render(struct grim_state *state, struct grim_box *geometry,
 		int output_flipped_y = output->screencopy_frame_flags &
 			ZWLR_SCREENCOPY_FRAME_V1_FLAGS_Y_INVERT ? -1 : 1;
 
-		cairo_surface_t *output_surface = cairo_image_surface_create_for_data(
-			buffer->data, cairo_fmt, buffer->width, buffer->height,
-			buffer->stride);
-		cairo_pattern_t *output_pattern =
-			cairo_pattern_create_for_surface(output_surface);
-
-		// All transformations are in pattern-local coordinates
-		cairo_matrix_t matrix;
-		cairo_matrix_init_identity(&matrix);
-		cairo_matrix_translate(&matrix,
-			(double)output->geometry.width / 2,
-			(double)output->geometry.height / 2);
-		cairo_matrix_scale(&matrix,
-			(double)raw_output_width / output_width,
-			(double)raw_output_height / output_height * output_flipped_y);
-		cairo_matrix_rotate(&matrix, -get_output_rotation(output->transform));
-		cairo_matrix_scale(&matrix, output_flipped_x, 1);
-		cairo_matrix_translate(&matrix,
-			-(double)output_width / 2,
-			-(double)output_height / 2);
-		cairo_matrix_translate(&matrix, -output_x, -output_y);
-		cairo_matrix_scale(&matrix, 1 / scale, 1 / scale);
-		cairo_pattern_set_matrix(output_pattern, &matrix);
-
-		cairo_pattern_set_filter(output_pattern, CAIRO_FILTER_BEST);
-
-		cairo_set_source(cairo, output_pattern);
-		cairo_pattern_destroy(output_pattern);
-
-		cairo_paint(cairo);
-
-		cairo_surface_destroy(output_surface);
+		pixman_image_t *output_image = pixman_image_create_bits(
```

Comment:
Do we need to call `pixman_image_create_bits_no_clear` instead here?

Reply:
No, the two functions only differ in whether they call `malloc` or `calloc` when `bits=NULL` is passed to them; see [pixman-bits-image.c#L1360](https://gitlab.freedesktop.org/pixman/pixman/-/blob/eadb82866b0f6a326a61c36f60e5c2be8f7479af/pixman/pixman-bits-image.c#L1360).